daemon.go

```diff
@@ -146,3 +147,33 @@ func (d *Daemon) listen() {
 		go d.handleConn(c)
 	}
 }
+
+func clearUnixSockets(path ma.Multiaddr) error {
+	c, _ := ma.SplitFirst(path)
+	if c.Protocol().Code != ma.P_UNIX {
+		return nil
+	}
+
+	if err := os.Remove(c.Value()); err != nil {
+		return err
+	}
+	
+	return nil
+}
+
+func (d *Daemon) Close() error {
+	if err := d.host.Close(); err != nil {
+		return err
```

Comment:
This will short-circuit if the host close fails and not clean up the unix sockets or the listener.
I think we should accumulate the errors in a `multierror` and proceed with the clean up regardless.

Reply:
I added multierror, if you don't mind checking that I implemented in a way you expected